### ISSUE TITLE
Custom OptimizerV2 frozen

### ISSUE THREAD

**Issue by OverLordGoldDragon** [tensorflow/tensorflow#33227]
I seek to implement [AdamW](https://github.com/OverLordGoldDragon/keras-adamw) for `OptimizerV2`, using syntax from [Keras](https://github.com/CyberZHG/keras-radam/blob/5bda32f5e26c6dc8140f7c6c95c13e9128341282/keras_radam/optimizer_v2.py) and [TF2](https://github.com/taki0112/RAdam-Tensorflow) implementations of RAdam - but am getting nowhere.  Below is a minimal reproducible example for a rewritten portion of [Adam](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/optimizer_v2/adam.py#L32); the complete version is just this revision, with `_prepare_local` deleted entirely (see [gist](https://gist.github.com/OverLordGoldDragon/aad492db1ff6f20a7983e600611799f7)). `print` statements for introspection. Observations:

 - Loss doesn't change at all (tested for 20 iters)
 - Weights don't change at all via `state_ops.assign_sub` (or via `.assign` w/ `math_ops.sub`)
 - `var_t` _does_ change, showing that subtracted value isn't negligibly small - but the change isn't applied to `var`
 - `var_t` changes slightly across iterations, possibly per GPU parallelism randomness (values do vary within 1%, not "almost zero")

<hr>

If linked implementations are correct, why is mine failing?

<hr>

```python
from tensorflow.keras.layers import Input, Dense
from tensorflow.keras.models import Model
from adam_v2 import Adam  # custom local module w/ described changes
reset_seeds()  # see below

ipt   = Input(shape=(4,))
x     = Dense(2, activation='relu', name='dense_1')(ipt)
out   = Dense(1, activation='sigmoid')(x)
model = Model(ipt, out)
model.compile(Adam(lr=1e-3), 'binary_crossentropy')

X = np.random.randn(32, 4)
Y = np.random.randint(0, 2, (32, 1))
print(model.train_on_batch(X,Y))
```
```python
<tf.Variable 'dense_1/kernel:0' shape=(4, 2) dtype=float32, numpy=
array([[-0.73267365,  0.13570619],
       [-0.00733662, -0.9662695 ],
       [ 0.00908351,  0.4974625 ],
       [-0.73932624, -0.32877016]], dtype=float32)>
      [[-0.7358344   0.13886832]
       [-0.00417447 -0.9631073 ]
       [ 0.00592144  0.49430028]
       [-0.73616403 -0.32560796]]
<tf.Variable 'dense_1/kernel:0' shape=(4, 2) dtype=float32, numpy=
array([[-0.73267365,  0.13570619],
       [-0.00733662, -0.9662695 ],
       [ 0.00908351,  0.4974625 ],
       [-0.73932624, -0.32877016]], dtype=float32)>

0.8960571  # loss
```

<hr>

```python
def _resource_apply_dense(self, grad, var):
    var_dtype = var.dtype.base_dtype
    lr_t = self._decayed_lr(var_dtype)
    m = self.get_slot(var, 'm')
    v = self.get_slot(var, 'v')
    beta_1_t = array_ops.identity(self._get_hyper('beta_1', var_dtype))
    beta_2_t = array_ops.identity(self._get_hyper('beta_2', var_dtype))
    epsilon_t = ops.convert_to_tensor(self.epsilon, var_dtype)
    
    m_t = state_ops.assign(m,
    		   beta_1_t * m + (1.0 - beta_1_t) * grad,
    		   use_locking=self._use_locking)
    v_t = state_ops.assign(v,
    		   beta_2_t * v + (1.0 - beta_2_t) * math_ops.square(grad),
    		   use_locking=self._use_locking)
    if self.amsgrad:
    	vhat = self.get_slot(var, 'vhat')
    	vhat_t = state_ops.assign(vhat, math_ops.maximum(vhat, v_t),
                                  use_locking=self._use_locking)
    	var_delta = m_t / (K.sqrt(vhat_t) + epsilon_t)
    else:
    	var_delta = m_t / (K.sqrt(v_t) + epsilon_t)
    var_t = math_ops.sub(var, lr_t * var_delta)

    if 'dense_1/kernel' in var.name:
    	print(var)
    	print(K.eval(var_t))
    var_update = state_ops.assign_sub(var, lr_t * var_delta,
    			              use_locking=self._use_locking)
    if 'dense_1/kernel' in var.name:
    	print(var)
    	
    updates = [var_update, m_t, v_t]
    if self.amsgrad:
    	updates.append(vhat_t)
    return control_flow_ops.group(*updates)
```
```python
import numpy as np
import random
import tensorflow as tf

def reset_seeds():
    np.random.seed(1)
    random.seed(2)
    if tf.__version__[0] == '2':
        tf.random.set_seed(3)
    else:
        tf.set_random_seed(3)
```

**Comment by OverLordGoldDragon** [comment]
**Solved**: change `K.sqrt` to `math_ops.sqrt`. Why it works: don't know.
